src/components/GitPanel.tsx

```diff
@@ -794,19 +794,30 @@ export class GitPanel extends React.Component<IGitPanelProps, IGitPanelState> {
    *
    * @param path - repository path
    */
-  private async _hasIdentity(path: string): Promise<void> {
-    // If the repository path changes, check the user identity
-    if (path !== this._previousRepoPath) {
+  private async _hasIdentity(path: string): Promise<string> {
+    // If the repository path changes or explicitly configured, check the user identity
+    if (
+      path !== this._previousRepoPath ||
+      this.props.settings.composite['promptUserIdentity']
+    ) {
       try {
         const data: JSONObject = (await this.props.model.config()) as any;
```

Comment:
I'ld like to avoid requesting the server every time.

Could you a new attribute to the `GitModel` - `lastAuthor: {name: string, email: string} | null`?

So calling `model.config()` will only happen if `lastAuthor` is `null`?